xe/thinCLIProtocol.cs

```diff
@@ -269,53 +269,63 @@ public static Stream DoRPC(TcpClient client, string method, Uri uri, Config conf
 
     public static class Types
     {
-	    public static uint unmarshal_int32(Stream stream){
+	    public static uint UnMarshalUint(Stream stream)
+        {
 		    uint a = (uint)stream.ReadByte();
 		    uint b = (uint)stream.ReadByte();
 		    uint c = (uint)stream.ReadByte();
 		    uint d = (uint)stream.ReadByte();
-		    //Console.WriteLine("a = " + a + " b = " + b + " c = " + c + " d = " + d);
 		    return (a << 0) | (b << 8) | (c << 16) | (d << 24);
 	    }
 
-        public static void marshal_int32(Stream stream, uint x){
+        public static void MarshalUint(Stream stream, uint x)
+        {
 		    uint mask = 0xff;
-		    stream.WriteByte((byte) ((x >> 0) & mask));
-		    stream.WriteByte((byte) ((x >> 8) & mask));
-		    stream.WriteByte((byte) ((x >> 16) & mask));
-		    stream.WriteByte((byte) ((x >> 24) & mask));
-	    }
+            stream.WriteByte((byte)((x >> 0) & mask));
+            stream.WriteByte((byte)((x >> 8) & mask));
+            stream.WriteByte((byte)((x >> 16) & mask));
+            stream.WriteByte((byte)((x >> 24) & mask));
+        }
 
-        public static int unmarshal_int(Stream stream){
-		    return (int)unmarshal_int32(stream);
+        public static int UnMarshalInt(Stream stream)
+        {
+		    return (int)UnMarshalUint(stream);
 	    }
 
-        public static void marshal_int(Stream stream, int x){
-		    marshal_int32(stream, (uint)x);
+        public static void MarshalInt(Stream stream, int x)
+        {
+		    MarshalUint(stream, (uint)x);
 	    }
 
-        public static byte[] unmarshal_n(Stream stream, uint n){
+        public static byte[] UnMarshalN(Stream stream, uint n)
+        {
 		    byte[] buffer = new byte[n];
-		    int toread = (int)n;
+            int toRead = (int)n;
 		    int offset = 0;
-		    while (toread > 0){
-			    int nread = stream.Read(buffer, offset, toread);
-			    offset= nread; toread -= nread;
-		    }
+
+		    while (toRead > 0)
+            {
+                int nRead = stream.Read(buffer, offset, toRead);
+                offset = nRead;
+                toRead -= nRead;
+            }
+
 		    return buffer;
 	    }
 
-        public static string unmarshal_string(Stream stream){
-		    uint length = unmarshal_int32(stream);
-		    byte[] buffer = unmarshal_n(stream, length);
+        public static string UnMarshalString(Stream stream)
+        {
+		    uint length = UnMarshalUint(stream);
+		    byte[] buffer = UnMarshalN(stream, length);
 		    Decoder decoder = Encoding.UTF8.GetDecoder();
 		    char[] chars = new char[decoder.GetCharCount(buffer, 0, (int)length)];
 		    decoder.GetChars(buffer, 0, (int)length, chars, 0);
 		    return new string(chars);
```

Comment:
nit yet again:
Looks like these files have a mixture of tab and spaces, which GitHub doesn't cope well with. Visual Studio's code formatting function will do it for you:
![image](https://user-images.githubusercontent.com/32554698/169056144-6d5498cf-b5a7-4a70-9aaa-81e1dbd52986.png)